manager/controllers/app/fybrikmodule_controller.go

```diff
@@ -0,0 +1,140 @@
+	"fybrik.io/fybrik/manager/controllers/utils"
+)
+
+// FybrikModuleReconciler reconciles a FybrikModule object
+type FybrikModuleReconciler struct {
+	client.Client
+	Name   string
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// Reconcile reconciles FybrikModule CRD
+// It receives FybrikModule CRD and selects the appropriate modules that will run
+// The outcome is a Plotter containing multiple Blueprints that run on different clusters
+func (r *FybrikModuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("FybrikModule", req.NamespacedName)
+	// obtain FybrikModule resource
+	moduleContext := &api.FybrikModule{}
+	if err := r.Get(ctx, req.NamespacedName, moduleContext); err != nil {
+		log.V(0).Info("The reconciled object was not found")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// If the object has a scheduled deletion time, update status and return
+	if !moduleContext.DeletionTimestamp.IsZero() {
+		// The object is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	observedStatus := moduleContext.Status.DeepCopy()
+	moduleVersion := moduleContext.GetGeneration()
+
+	// check if module has been validated before or if validated module is outdated
+	if string(moduleContext.Status.ValidModule) == "" || observedStatus.ValidatedGeneration != moduleVersion {
+		// do validation on moduleContext
+		err := ValidateFybrikModule(moduleContext, "/tmp/taxonomy/fybrik_module.json")
+		log.V(0).Info("Reconciler validating Fybrik module")
+		moduleContext.Status.ValidatedGeneration = moduleVersion
+		// if validation fails
+		if err != nil {
+			// set error message
+			log.V(0).Info("Fybrik module validation failed " + err.Error())
+			moduleContext.Status.ErrorMessage = err.Error()
+			moduleContext.Status.ValidModule = v1.ConditionFalse
+		} else {
+			moduleContext.Status.ValidModule = v1.ConditionTrue
+			moduleContext.Status.ErrorMessage = ""
+		}
+	}
+
+	// Update CRD status in case of change (other than deletion, which was handled separately)
+	if !equality.Semantic.DeepEqual(&moduleContext.Status, observedStatus) && moduleContext.DeletionTimestamp.IsZero() {
+		log.V(0).Info("Reconcile: Updating status for desired generation " + fmt.Sprint(moduleContext.GetGeneration()))
+		if err := r.Client.Status().Update(ctx, moduleContext); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+// NewFybrikModuleReconciler creates a new reconciler for FybrikModules
+func NewFybrikModuleReconciler(mgr ctrl.Manager, name string) *FybrikModuleReconciler {
+	return &FybrikModuleReconciler{
+		Client: mgr.GetClient(),
+		Name:   name,
+		Log:    ctrl.Log.WithName("controllers").WithName(name),
+		Scheme: mgr.GetScheme(),
+	}
+}
+
+func ValidateFybrikModule(module *api.FybrikModule, taxonomyFile string) error {
+	var allErrs []*field.Error
+
+	// Convert Fybrik module Go struct to JSON
+	applicationJSON, err := json.Marshal(module)
```

Comment:
Maybe moduleJSON instead of applicationJSON

Reply:
Fixed